### PR TITLE
feat: allow ability to filter out joinOptions in the TagsJoinPicker and TagsQuerySearchBar

### PR DESCRIPTION
**Pull request by ccalbaugh**
You can now filter down the ['none', 'all', 'any'] options TagsJoinPicker by passing an explicit array of options.  

EX: `joinOptions: ['all', 'any']`
Or in TagsQuerySearchBar it would be `tagsQueryProps={{ joinOptions: ['any', 'all'] }}`

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1b7da6b02b91767349948adaf01cdbcf26b85f57
</p>

File: src/exampleTypes/TagsJoinPicker.js [hunk 1]
```diff
@@ -10,17 +10,25 @@ export let tagToGroupJoin = (x = 'any') =>
...
 ]
 
-let TagsJoinPicker = ({ node, tree, theme: { Select } }) => (
+let getJoinOptions = options =>
+  _.size(options)
+    ? _.filter(
+        option => _.includes(_.get('value', option), options),
+        defaultJoinOptions
+      )
+    : defaultJoinOptions
```
Comment (stellarhoof) [inline src/exampleTypes/TagsJoinPicker.js:25]: `_.intersectionWith((left, right) => left.value === right, defaultJoinOptions)` ;)

**Review by stellarhoof — CHANGES_REQUESTED** [review]

File: src/exampleTypes/TagsJoinPicker.js (hunk 1 above)
Comment (daedalus28) [reply src/exampleTypes/TagsJoinPicker.js:25]: Nice ⛳

**Review by stellarhoof — APPROVED** [review]

**Pull request merged**